### PR TITLE
Enhancements/devel mode deployment agent and varia 🚧🚧🚧

### PR DESCRIPTION
**Pull request by mrnicegyu11**
This PR:
- ♻️ Bumps deployment agent to `0.11.0`
- ♻️ Improves workflow with devel builds of the deployment agent
- ♻️ Adds some logs
- ✨ Adds Makefile targets to create release-tags

🚧🚧🚧:

- Requires deployment agent version 0.11.0, see https://github.com/ITISFoundation/osparc-deployment-agent/pull/156
- Needs to be merged with https://git.speag.com/oSparc/osparc-ops-deployment-configuration/-/merge_requests/3

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -21,7 +21,9 @@ certificates/domain.key:
 
 .PHONY: .create-secrets
 .create-secrets:
-	@$(MAKE_C) certificates deploy
+	@echo "Creating docker secrets..." && \
```
Comment (sanderegg) [inline Makefile:24]: minor: ```#Creating docker secrets...``` putting it as a comment does the same as your echo here. it makes it slightly more readable.

```bash
# Creating docker secrets...
@$(MAKE_C) certificates deploy
# Done: Creating docker secrets
```

**Review by sanderegg — APPROVED** [review]

File: Makefile (hunk 1 above)
Comment (mrnicegyu11) [reply Makefile:24]: thanks! will address

**Review by GitHK — APPROVED** [review]

File: Makefile [hunk 2]
```diff
@@ -166,3 +168,27 @@ reset-prune: ## resets docker swarm, removes all images, volumes, secrets, netwo
...
+staging_prefix := staging_
+release_prefix := v
+_git_get_current_branch = $(shell git rev-parse --abbrev-ref HEAD)
+_git_get_formatted_staging_tag = ${staging_prefix}${name}$(version)
+_git_get_formatted_release_tag = ${release_prefix}$(version)
+
+# NOTE: be careful that GNU Make replaces newlines with space which is why this command cannot work using a Make function
+
+.PHONY: release-prod
+release-prod: ## Helper to create a staging or production release in Github (usage: make release-prod version=1.2.3)
```
[the remark this answers is not in the document]
Comment (mrnicegyu11) [reply Makefile:185]: correct

File: services/deployment-agent/Makefile [hunk 3]
```diff
@@ -73,7 +73,9 @@ ${TEMP_COMPOSE}: .env $(docker-compose-configs)
 
 .PHONY: ${TEMP_COMPOSE-devel}
 ${TEMP_COMPOSE-devel}: .env $(docker-compose-configs)
-	@cat docker-compose.yml | $(_yq) '. *+ load("docker-compose.devel.yml")' > $@
+	@cat docker-compose.yml | $(_yq) '. *+ load("docker-compose.devel.yml")' > .temp$@ && \
+	docker-compose --file .temp$@ --log-level=ERROR config > $@; \
```
[the remark this answers is not in the document]
Comment (mrnicegyu11) [reply services/deployment-agent/Makefile:77]: will add, thx

File: services/deployment-agent/deployment_config.template.yml [hunk 4]
```diff
@@ -27,13 +28,15 @@ main:
...
       pull_only_files: false
+      ${TAGS_SIMCORE_REPO}
       username: ${GIT_OPS_LOGIN}
       password: ${GIT_OPS_PASSWORD}
       paths: []
     - id: simcore-ops-config-repo
       url: ${GIT_DEPLOYEMENTCONFIG_REPO_URL}
       branch: ${GIT_DEPLOYEMENTCONFIG_REPO_BRANCH}
-      pull_only_files: true
+      ${TAGS_DEPLOYMENT_CONFIGURATION_REPO}
```
[the remark this answers is not in the document]
Comment (mrnicegyu11) [reply services/deployment-agent/deployment_config.template.yml:38]: `TAGS_DEPLOYMENT_CONFIGURATION_REPO="tags: v1.*"` for example

**Pull request merged**